tests/platform/cloud/AcceptanceTests_oauth2_privilegedPrefsCreator.js

```diff
@@ -56,16 +57,103 @@ gpii.tests.cloud.oauth2.addPrefs.cleanUpTmpFiles = function () {
     gpii.tests.cloud.oauth2.addPrefs.filesToDelete.length = 0;
 };
 
+gpii.tests.cloud.oauth2.addPrefs.sendAccessTokenRequest = function (accessTokenRequest, options) {
+    var formBody = {
+        grant_type: "client_credentials",
+        scope: options.scope,
+        client_id: options.client_id,
+        client_secret: options.client_secret
+    };
+
+    gpii.test.cloudBased.oauth2.sendRequest(accessTokenRequest, options, formBody, "accessTokenForm");
+};
+
+gpii.tests.cloud.oauth2.addPrefs.sendAddPrefsRequest = function (securedAddPrefsRequest, prefsData, secured, accessToken, view) {
+    view = view || "flat";
+    secured = secured || true;
+
+    var options = {
+        path: "/add-preferences?view=" + view
+    };
+
+    var securedHeader = {
+        headers: {
+            Authorization: "Bearer " + accessToken,
+            "Content-Type": "application/json"
+        }
+    };
+
+    var securedOptions = fluid.extend({}, options, securedHeader);
+
+    options = secured ? securedOptions : options;
+
+    securedAddPrefsRequest.send(prefsData, options);
+};
+
+gpii.tests.cloud.oauth2.addPrefs.verifyAddPrefsResponse = function (body, addPrefsRequest, expectedPrefs, prefsDir) {
+    var response = gpii.test.verifyJSONResponse(body, addPrefsRequest);
+    jqUnit.assertValue("Should have received a user token", response.userToken);
+    jqUnit.assertDeepEq("The received saved preferences data should be expect", expectedPrefs, response.preferences);
+    addPrefsRequest.userToken = response.userToken;
+
+    var filePath = path.resolve(prefsDir, response.userToken) + ".json";
+    gpii.tests.cloud.oauth2.addPrefs.filesToDelete.push(filePath);
+};
+
+gpii.tests.cloud.oauth2.addPrefs.sendGetPrefsRequest = function (getPrefsRequest, addPrefsRequest, view) {
+    view = view || "flat";
+    var options = {
+        path: "/preferences/" + addPrefsRequest.userToken + "?view=" + view
+    };
+
+    getPrefsRequest.send(null, options);
+};
+
+gpii.tests.cloud.oauth2.addPrefs.verifyGetPrefsResponse = function (body, getPrefsResponse, expectedPrefs) {
+    var response = gpii.test.verifyJSONResponse(body, getPrefsResponse);
+    jqUnit.assertDeepEq("The received saved preferences data should be expect", expectedPrefs, response);
+};
+
 fluid.defaults("gpii.tests.cloud.oauth2.addPrefs.cleanupTmpFiles", {
     gradeNames: "fluid.component",
     listeners: {
         "onDestroy.cleanupTmpFiles": "gpii.tests.cloud.oauth2.addPrefs.cleanUpTmpFiles"
     }
 });
 
+fluid.defaults("gpii.tests.cloud.oauth2.addPrefs.requests", {
+    gradeNames: ["fluid.component"],
+    components: {
+        addPrefsRequest: {
+            type: "kettle.test.request.http",
+            options: {
+                // path: "/add-preferences?view=%view",  // Cannot dynamically config path
```

Comment:
The existing kettle.test.request.http can cope fine with this templating - you can add the path as listed here and then send {termMap: {view: view}} as the "directOptions" argument